### ISSUE TITLE
Spyder gives wrong information about PEP8 with annotations

### ISSUE THREAD

**Issue by martmists** [spyder-ide/spyder#4314]
## Description of your problem
Spyder recommends no whitespace around kwargs when there are type annotations, but according to PEP 8 there should be:

*When combining an argument annotation with a default value, use spaces around the = sign (but only for those arguments that have both an annotation and a default).*

*Yes:*

```python
def munge(sep: AnyStr = None): ...
def munge(input: AnyStr, sep: AnyStr = None, limit=1000): ...
```
*No:*

```python
def munge(input: AnyStr=None): ...
def munge(input: AnyStr, limit = 1000): ...
```

**What steps will reproduce the problem?**

```python
def func(arg, kwarg: str=None):  # Good according to spyder
    return "This is bad according to PEP 8"

def func(arg, kwarg: str = None):  # Shows an exclamation mark
    return "This is good according to PEP 8"
```

## Versions and main components

* Spyder Version: 3.1.3
* Python Version: 3.6.0
* Qt Version: Unknown
* PyQt Version: 5.8.1
* Operating system: Windows 10 Home build 14393.rs1_release_inmarket.170303-1614

**Comment by ccordoba12** [comment]
This is not under Spyder control, i.e. Spyder only parses and reports warnings generated by the `pep8`, and once 3.2 is released, `pycodestyle`, packages.

So I think you should report this issue in the `pycodestyle` repository.

**Comment by martmists** [comment]
After checking the file with `pycodestyle file.py`, it seems nor `pycodestyle`, nor `flake8` nor `pylint` cause that error in that file, despite that error still being there in Spyder.

**Comment by rlaverde** [comment]
If you put the mouse over the exclamation marker what is the message?

Maybe It was generated for `pyflakes` (the example that you mention will cause a warning for the redefiinition of `func`)

**Comment by martmists** [comment]
`E251 Unexpected spaces around keyword / parameter equals`

**Comment by martmists** [comment]
Just did a quick check, the error is thrown by the `pep8` module. I'm assuming this issue is deprecated then and I'll just have to wait for `pep8` to be replaced by `pycodestyle` as soon as possible.

**Comment by rlaverde** [comment]
Ohh I see, so this error is related to #3197, that have already been fixed, and will be part of Spyder 3.2

**Comment by martmists** [comment]
The 3.2 milestone was planned for the day after tomorrow? I somehow feel like they're not going to reach it ;) Until then I'll just ignore this error and use pycodestyle manually I guess
